hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -590,6 +596,10 @@ public HRegionServer(final Configuration conf) throws IOException {
       this.abortRequested = false;
       this.stopped = false;
 
+      // initiate online slowlog ringbuffer only for RegionServers
+      if (!(this instanceof HMaster)) {
```

Comment:
If we don't initialize this and if user tries to retrieve get_slow_responses, request can't go to `SlowLogRecorder`. We should let the request go there and return empty response since queue is empty. And when user realize that he is getting empty response because he has not enabled required config, he will enable it, in fact he can enable it dynamically. Dynamic config support is also on the way in follow up task. So, we should initialize this object and let the object determine if queue is to be created based on config and if so, what is the size limit of queue etc.
`HRegionServer` is not the one to decide queue creation, `SlowLogRecorder` is the one :)